scripts/backup_with_rotation_original.sh

```
@@ -0,0 +1,47 @@
+    # Create backup directory and zip its contents
+    zip -r "${backup_dir}.zip" "$source_dir" >/dev/null
+    if [ $? -eq 0 ]; then
+        echo "Backup created successfully: ${backup_dir}.zip"
+    else
+        echo "Error: Failed to create backup."
+    fi
+}
+
+# Function to perform rotation and keep only the last 3 backups
+function perform_rotation {
+    local backups=($(ls -t "${source_dir}/backup_"*.zip 2>/dev/null))
```

Shellcheck

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)